ldp/graph/common_ops.py

```diff
@@ -283,6 +297,35 @@ async def forward(
 
         return result.messages[0]
 
+    async def _call_single_and_maybe_validate(
+        self, model: LLMModel, num_retries: int, **kwargs
+    ) -> LLMResult:
+        if not self.response_validator:
+            # If a response validator is not supplied, then we should not do any retries here - leave
+            # that for LiteLLM to handle.
+            return await model.call_single(**kwargs)
+
+        @tenacity.retry(
+            retry=tenacity.retry_if_exception_type(ResponseValidationError),
+            # num_retries+1 because the first call is not a retry
+            stop=tenacity.stop_after_attempt(num_retries + 1),
+            wait=tenacity.wait_fixed(1),
+        )
+        async def call_and_validate() -> LLMResult:
+            result = await model.call_single(**kwargs)
```

Comment:
Can you add a `# NOTE:` or something somewhere talking about how we can get many retries here:
- LiteLLM retrying some things
- `response_validator` leading to retrying other things

The max amount of retries can now be `num_retries ** 2`, right?